### PR TITLE
update: add weak-compatible to dependency check

### PR DESCRIPTION
**Pull request by Two-Hearts**
This PR is based on this discussion: https://github.com/apache/skywalking/discussions/11391.
A new list called `weak-compatible` is added to the [skywalking-eyes compatibility list](https://github.com/apache/skywalking-eyes/blob/main/assets/compatibility/Apache-2.0.yaml). 
A new flag called `weak-compatible` is added to the `dependency check` command.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Thanks for raising this. Please fix the CI first.

**Comment by Two-Hearts** [comment]
Working on the CI, looks like a lint problem.

File: commands/deps_check.go [hunk 1]
```diff
@@ -26,6 +26,12 @@ import (
 	"github.com/apache/skywalking-eyes/pkg/deps"
 )
 
+var weakCompatible bool
+
+func init() {
+	DepsCheckCommand.PersistentFlags().BoolVarP(&weakCompatible, "weak-compatible", "w", false, "if set, include the weak-compatible list in dependencies checking")
```
Comment (kezhenxu94) [inline commands/deps_check.go:32]: To me, "include the weak-compatible list in dependencies checking" doesn't indicate the "checking" accepts or rejects the list in dependencies, which might be ambiguous for users, what do you think if we make it more explicit in the description, and add a note, for example:

if set to true, treat the `weak-compatible` licenses as compatible in dependencies check. Note: when set this to true, make sure to manually confirm that you are using the `weak-compatible` licenses under the required conditions.

**Review by kezhenxu94 — COMMENTED** [review]
A tiny suggestion about the description of the new command option, others look good to me, LGTM after CI is passed

File: commands/deps_check.go (hunk 1 above)
Comment (Two-Hearts) [reply commands/deps_check.go]: Thanks @kezhenxu94, I updated the description based on your example. CI fixed.

**Review by wu-sheng — APPROVED** [review]
Good to have this. Wait for @kezhenxu94 on codes.

**Review by kezhenxu94 — APPROVED** [review]
Thank you! @Two-Hearts

**Pull request merged**